### PR TITLE
Split RequireCampaignSlugcat from SlugcatCustomToggle

### PR DESCRIPTION
**Pull request by invalidunits**
Splits RequireCampaignSlugcat configuration from SlugcatCustomToggle. So host can choose to use unstable campaigns without allowing clients to select their own slugcats.

### CODE REVIEW

**Comment by 6fears7** [comment]
We will need to sync the choice to clients as well. 

If I'm understanding the PR correctly, you want to support scenarios where the host only wants to have clients who join to match the current campaign slugcat, period.

**Comment by 6fears7** [comment]
Pinging @youbitchoc for any thoughts regarding this.

**Comment by youbitchoc** [comment]
lgtm

**Comment by youbitchoc** [comment]
thoughts on putting the toggle in storymenu instead? so it's easier to enable/disable when selecting campaigns

**Comment by 6fears7** [comment]
Yes this is my preference. It would be easier to do a sync on any checkboxes detected in the menu

**Comment by youbitchoc** [comment]
> do a sync on any checkboxes detected in the menu

special onlinecheckbox class? sure
by default? no

not worth the effort imo, maybe when modders start building on meadow

**Comment by 6fears7** [comment]
While having this be a checkbox in story menu makes sense based on accessibility, we only have one more natural spot for a box to go (below friendly fire toggle) before we move more elsewhere. 

Additionally, this box context does fit with the "parent" (and temporary) remix setting of choosing other campaigns. 

Knowing this, do you still want to put it in story menu or leave it in remix?

**Comment by youbitchoc** [comment]
> we only have one more natural spot for a box to go (below friendly fire toggle) before we move more elsewhere

future popup menu
if a more important checkbox comes up then we can just swap it out

> this box context does fit with the "parent" (and temporary) remix setting of choosing other campaigns

enabling modded scugs is the client's personal choice, forcing campaign scug is the host's per-session choice

**Comment by 6fears7** [comment]
> enabling modded scugs is the client's personal choice, forcing campaign scug is the host's per-session choice

Argument works for me. @invalidunits please movie the checkbox to the StoryOnlineMenu below the Friendly Fire checkbox.

**Review by 6fears7 — CHANGES_REQUESTED** [review]
Move checkbox into Story Menu below FF checkbox

**Comment by invalidunits** [comment]
@6fears7 This should be ready for merging.

File: Game/MeadowRemixOptions.cs [hunk 1]
```diff
@@ -108,10 +110,18 @@ public override void Initialize()
...
-                new OpCheckBox(SlugcatCustomToggle, new Vector2(10f, 320)),
-                new OpLabel(40f, 320, RWCustom.Custom.ReplaceLineDelimeters("If selected, hosts can choose slugcat campaigns that are unstable. <LINE>Clients can choose their own Slugcats inside a host's Story campaign"))
+                new OpCheckBox(SlugcatCustomToggle, new Vector2(160f, 350)),
+
+                new OpLabel(40f, 320, RWCustom.Custom.ReplaceLineDelimeters("If selected, hosts can choose slugcat campaigns that are unstable."))
+                {
+                    verticalAlignment = OpLabel.LabelVAlignment.Center
+                },
+
+                new OpLabel(10f, 290, "Require Campaign Slugcat:", bigText: false),
```
Comment (6fears7) [inline Game/MeadowRemixOptions.cs:122]: Please remove traces of remix bool due to move in story menu

**Review by 6fears7 — CHANGES_REQUESTED** [review]
Please move the checkbox

File: Game/MeadowRemixOptions.cs (hunk 1 above)
Comment (invalidunits) [reply Game/MeadowRemixOptions.cs]: This should be resolved now.

**Comment by youbitchoc** [comment]
has this been tested to work?

**Review by 6fears7 — APPROVED** [review]
Tested to work.

**Review by 6fears7 — APPROVED** [review]

**Pull request merged**